datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -643,18 +643,25 @@ impl OptimizerRule for PushDownFilter {
                 for expr in predicates {
                     let columns = expr.to_columns()?;
                     if columns.is_empty()
-                        || !columns
-                            .intersection(&used_columns)
-                            .collect::<HashSet<_>>()
-                            .is_empty()
```

Comment:
The original performance was bad

Reply:
As I mentioned in the last PR, I think we do not need to check the aggregate Exprs, but just check the group by Exprs. In some cases, the same column can exist in both aggregate Exprs and group by Exprs, for example `select count(distinct col_a),  col_a from table  group by col_a; ` .  If there is a Filter applied to col_a, the Filter can still be pushed down even it is referred  by the agg Exprs.

Reply:
The logic should check all the columns used by the Filter predicate is the subset of the group by Exprs output Columns.

Reply:
Yes. For push_down_filter through Agg, we can push `Expr` in `groupby_expr`.
Has add it.